### PR TITLE
Add Adw1 and Gdk4

### PR DESCRIPTION
**Pull request by CharlieQLe**
Add the Adw module and upgrade Gdk to 4.0

### CODE REVIEW

**Comment by lovetox** [comment]
You need to move the current Gdk file into the __multiversion folder and rename it to Gdk3

**Comment by CharlieQLe** [comment]
:+1: Did that, and I also added most of the types for Adw, based on the documentation at [https://amolenaar.github.io/pgi-docgen/](https://amolenaar.github.io/pgi-docgen/).

File: gi-stubs/repository/Adw.pyi [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+
+class AnimationTarget: ...
+
+class Application:
+    def get_style_manager(self) -> StyleManager: ...
+    
+
+class ApplicationWindow:
+    def get_content(self) -> Gtk.Widget: ...
+    def set_content(self, widget: Gtk.Widget | None) -> None: ...
```
Comment (lovetox) [inline gi-stubs/repository/Adw.pyi:107]: the union notation was added in python 3.10 i believe, we currently only require 3.7.

Can you use Union[] instead?

Comment (CharlieQLe) [reply gi-stubs/repository/Adw.pyi:107]: Ah, I'll start working on that.

Comment (CharlieQLe) [reply gi-stubs/repository/Adw.pyi]: The union notation should all be replaced by Union[] now.

**Comment by lovetox** [comment]
Can you rebase your changes please

**Comment by lovetox** [comment]
Please use `black` to format the files

**Comment by CharlieQLe** [comment]
Should be good now.

**Comment by lovetox** [comment]
Are you sure its rebased? it still shows `This branch cannot be rebased due to conflicts` for me

**Comment by CharlieQLe** [comment]
Just rebased it again, since there was an edit to the README.

**Comment by CharlieQLe** [comment]
Looks like running mypy with the existing example fails because of `Gdk.Atom`. Would removing it and using a different class under `Gdk` be a sufficient fix?

**Comment by lovetox** [comment]
Can you please squash all commit into one and prefix the commit message with "typing: "

**Comment by lovetox** [comment]
> Looks like running mypy with the existing example fails because of `Gdk.Atom`. Would removing it and using a different class under `Gdk` be a sufficient fix?

Yes please fix the examples as you see fit

**Pull request closed without merging**